File: app/src/main/java/com/pitchai/flickrrecentphotobrowser/data/FlickrRestApiService.java
package com.pitchai.flickrrecentphotobrowser.data;

import com.pitchai.flickrrecentphotobrowser.data.photoinfo.PhotoDetails;
import com.pitchai.flickrrecentphotobrowser.data.photoinfo.PhotoInfo;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * Created by pgrajama on 12/10/16.
 */

/**
 * Rest API interface
 */
public interface FlickrRestApiService {
    @GET("/services/rest")
    Observable<PhotoInfo> getRecentPhotoList(@Query("method") String method,
                                             @Query("nojsoncallback") String callback,
                                             @Query("format") String format,
                                             @Query("api_key") String apiKey,
                                             @Query("page") int requestPageNumber,
                                             @Query("extras") String extras);

    @GET("/services/rest")
    Observable<PhotoDetails> getPhotoInfo(@Query("method") String method,
                                          @Query("nojsoncallback") String callback,
                                          @Query("format") String format,
                                          @Query("api_key") String apiKey,
                                          @Query("photo_id") String photoID);


}

File: app/src/main/java/com/pitchai/flickrrecentphotobrowser/data/FlickrRestApiService.kt
package com.pitchai.flickrrecentphotobrowser.data

import com.pitchai.flickrrecentphotobrowser.data.photoinfo.PhotoDetails
import com.pitchai.flickrrecentphotobrowser.data.photoinfo.PhotoInfo
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * Created by pgrajama on 12/10/16.
 */
/**
 * Rest API interface
 */
interface FlickrRestApiService {
    @GET("/services/rest")
    fun getRecentPhotoList(@Query("method") method: String?,
                           @Query("nojsoncallback") callback: String?,
                           @Query("format") format: String?,
                           @Query("api_key") apiKey: String?,
                           @Query("page") requestPageNumber: Int,
                           @Query("extras") extras: String?): Observable<PhotoInfo?>?

    @GET("/services/rest")
    fun getPhotoInfo(@Query("method") method: String?,
                     @Query("nojsoncallback") callback: String?,
                     @Query("format") format: String?,
                     @Query("api_key") apiKey: String?,
                     @Query("photo_id") photoID: String?): Observable<PhotoDetails?>?
}